src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1370,13 +1368,8 @@ void Connection::receive_query_error_message(int raw_error_code, std::string_vie
                                             "Received a FLX query error message on a non-FLX sync connection"});
     }
 
-    Session* sess = find_and_validate_session(session_ident, "QUERY_ERROR");
-    if (REALM_UNLIKELY(!sess)) {
-        return;
-    }
-
-    if (auto status = sess->receive_query_error_message(raw_error_code, message, query_version); !status.is_ok()) {
-        close_due_to_protocol_error(std::move(status));
+    if (Session* sess = find_and_validate_session(session_ident, "QUERY_ERROR")) {
+        sess->receive_query_error_message(raw_error_code, message, query_version);
```

Comment:
receive_query_error_message() only ever returned Status::Ok() the code handling it returning an error was dead.